components/omega/test/ocn/EosTest.cpp

```diff
@@ -0,0 +1,494 @@
+      Err++;
+      LOG_ERROR("EosTest: error initializing parallel IO");
+   }
+
+   int DecompErr = Decomp::init(mesh);
+   if (DecompErr != 0) {
+      Err++;
+      LOG_ERROR("EosTest: error initializing default decomposition");
+   }
+
+   int MeshErr = HorzMesh::init();
+   if (MeshErr != 0) {
+      Err++;
+      LOG_ERROR("EosTest: error initializing default mesh");
+   }
+
+   const auto &Mesh = HorzMesh::getDefault();
+   std::shared_ptr<Dimension> VertDim =
+       Dimension::create("NVertLevels", NVertLevels);
+   return Err;
+}
+
+int testEosMapping() {
+   int Err = 0;
+   // test initialization
+   int EosErr = Eos::init();
+   if (EosErr != 0) {
+      Err++;
+      LOG_ERROR("EosTest: error initializing default Eos");
+   }
+
+   // test retrievel of default
+   Eos *DefEos = Eos::getDefault();
+
+   if (DefEos) {
+      LOG_INFO("EosTest: Default Eos retrieval PASS");
+   } else {
+      Err++;
+      LOG_INFO("EosTest: Default Eos retrieval FAIL");
+      return -1;
+   }
+
+   const auto *Mesh = HorzMesh::getDefault();
+   // test creation of another Eos
+   Eos::create("TestEos", Mesh, NVertLevels);
+
+   if (Eos::get("TestEos")) {
+      LOG_INFO("EosTest: Non-default Eos retrieval PASS");
+   } else {
+      Err++;
+      LOG_INFO("EosTest: Non-default Eos retrieval FAIL");
+   }
+
+   // test erase
+   Eos::erase("TestEos");
+
+   if (Eos::get("TestEos")) {
+      Err++;
+      LOG_INFO("EosTest: Non-default Eos erase FAIL");
+   } else {
+      LOG_INFO("EosTest: Non-default Eos erase PASS");
+   }
+
+   // Eos::clear();
+
+   return Err;
+}
+
+int testEosLinear() {
+   int Err          = 0;
+   const auto *Mesh = HorzMesh::getDefault();
+   // create Eos to test
+   Eos::create("LinearEos", Mesh, NVertLevels);
+   Eos *TestEos       = Eos::get("LinearEos");
+   TestEos->EosChoice = EosType::Linear;
+
+   // create ocean state array
+   Array2DReal Sarray = Array2DReal("Sarray", Mesh->NCellsAll, NVertLevels);
+   Array2DReal Tarray = Array2DReal("Tarray", Mesh->NCellsAll, NVertLevels);
+   Array2DReal Parray = Array2DReal("Parray", Mesh->NCellsAll, NVertLevels);
+   // Use Kokkos::deep_copy to fill the entire view with the ref value
+   deepCopy(Sarray, Sa);
+   deepCopy(Tarray, Ct);
+   deepCopy(Parray, P);
+   deepCopy(TestEos->SpecVol, 0.0);
+
+   // Key calculation
+   TestEos->computeSpecVol(TestEos->SpecVol, Tarray, Sarray, Parray);
+   // LOG_INFO("EosTest, testEosLinear: produced SpecVol from Eos");
+
+   // check on all array values
+   int numMismatches = 0;
+   Kokkos::parallel_reduce(
+       "CheckSpecVolMatrix-linear",
+       Kokkos::MDRangePolicy<Kokkos::Rank<2>>({0, 0},
+                                              {Mesh->NCellsAll, NVertLevels}),
+       KOKKOS_LAMBDA(const int i, const int j, int &localCount) {
+          if (!isApprox(TestEos->SpecVol(i, j), LinearExpValue, RTol)) {
+             localCount++;
+          }
+       },
+       numMismatches);
+
+   bool allMatch = (numMismatches == 0);
```

Comment:
Don't need the intermediate bool here. Just use `if (numMismatches != 0)`

Reply:
Done!